src/js/editor/editor.js

```diff
@@ -307,20 +299,27 @@ class Editor {
 
   handleDeletion(event) {
     event.preventDefault();
+    const key = Key.fromEvent(event),
+          direction = key.direction;
 
     let offsets = this.cursor.offsets;
     let currentMarker, currentOffset;
-    this.run((postEditor) => {
+
+    this.run(postEditor => {
       let results;
       if (this.cursor.hasSelection()) {
         results = postEditor.deleteRange(offsets);
       } else {
-        // FIXME: perhaps this should accept this.cursor.offsets?
-        results = postEditor.deleteCharAt(offsets.headMarker, offsets.headOffset-1);
+        const deletePosition = {
+          marker: offsets.headMarker,
+          offset: offsets.headOffset
+        };
+        results = postEditor.deleteFrom(deletePosition, direction);
```

Comment:
Yeah, I think directional deletion makes something like `deleteCharAt` feel awkward. If we are deleting, the direction matters. And we want to delete __from__ a cursor position rather than __at__ a character offset (that's why I changed the method name from `deleteCharAt` to `deleteFrom`). For instance to backward-delete "__at__" the beginning of a section (joining with the previous section) is semantically different than deleting the character "at" the 0th position of the section (in that case, at least to my ears, no joining would be implied, just removal of the character).

I could see an argument for a `deleteCharAt` method that had no implication of joining anything and only removes a character in a marker at a given character offset. I'd consider `removeCharAt(marker, offset)` to be a better name, though.

I still have a little trouble following the many conditionals of LL's `insertBefore` — I would be hesitant to combine the forward- and backward- deletion cases into a single method for fear that the logic would be tortuous. They have a lot of symmetric similarity, but only a little duplicated code (when the cursor offset is in the middle of a marker — that would be a good candidate to refactor into a `removeCharAt` method), and there's a real difference between them in that the `nextCursor` marker and offset do not need to change (except one edge case) for forward-deletion but they do in nearly every case for backward-deletion.